hooks/pre/31-puppet_agent_oauth.rb

```diff
@@ -3,6 +3,13 @@
     unless execute("/opt/puppetlabs/puppet/bin/ruby -e \"require 'oauth'\"", false, false)
       success = execute("yum -y reinstall puppet-agent-oauth", false, true)
 
+      logger.error("Failed to reinstall puppet-agent-oauth. Please check that the package is available from a repository.") unless success
+    end
+  end
+  if File.exist?('/opt/puppetlabs/puppet/bin/ruby') && execute("dpkg-query --show --showformat='${db:Status-Status}' puppet-agent-oauth | grep -q '^installed'", false, false)
```

Comment:
Perhaps it can be formatted different. You are allowed to define functions in hooks so perhaps there should be something `def package_is_installed?(package)` that returns a boolean? Similar, a `reinstall_package(package)` can be good. You do have `facts` as a variable with the output of `facter` that can help you if needed.